src/DependencyInjection/Compiler/TwigStringExtensionCompilerPass.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\PageBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Twig\Extra\String\StringExtension;
+
+final class TwigStringExtensionCompilerPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        foreach ($container->findTaggedServiceIds('twig.extension') as $id => $attributes) {
+            if (StringExtension::class === $container->getDefinition($id)->getClass()) {
```

Comment:
I just see one case escaping from this condition. I've hit that scenario some time ago in other package and had to check that case too, but I'm not able to remember where.
The fact is, `$container->getDefinition($id)->getClass()` can return a parameter name (like "%class%"). So, this condition should take care about this situation making the conversion (`ContainerBuilder::getParameter()`) before the current check.

Reply:
Technicly it is possible but it will be very strange to use final class in this case.